app/javascript/components/gtl/DataTable.jsx

```diff
@@ -120,10 +132,17 @@ export const DataTable = ({
               })}
             >
               { row.cells[columnKey].is_checkbox && !settings.hideSelect && !inEditMode() &&
-                <input
-                  onChange={ev => localOnItemSelected(row, ev)}
+              <label className="hiddenCheckboxLabel" id={`check_${row.id}`} aria-hidden="true">{`check_${row.id}`}</label> &&
+              <input
+                  onChange={localOnItemSelected(row)}
+                  onKeyPress={localOnItemSelected(row)}
+                  role="checkbox"
+                  aria-checked="false"
+                  tabIndex="0"
+                  aria-labelledby={`check_${row.id}`}
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

The `aria-labelledby` should point to a different field that has the ID set in here. It supposed to be a reverse link to a label, the opposite of `<label for=`